rfc/text/advanced/ap_27_pubsub_event_history.md

```diff
@@ -1,58 +1,132 @@
-2. Can `wamp.topic.history.after` be implemented (efficiently) at all?
-3. How does that interact with pattern-based subscriptions?
-4. The same question as with the subscriber lists applies where: to stay within our separation of roles, we need a broker + a separate peer which implements the callee role. Here we do not have a mechanism to get the history from the broker.
-5. How are black/whitelisted sessionIDs treated? A client which requests event history will have a different sessionID than on previous connections, and may receive events for which it was excluded in the previous session, or not receive events for which it was whitelisted. - see https://github.com/wamp-proto/wamp-proto/issues/206
-
-
-**Feature Announcement**
-
-Support for this feature MUST be announced by *Subscribers* (`role := "subscriber"`) and *Brokers* (`role := "broker"`) via
+The results of all RPC above are the same and looks like an `arguments` array of `Event` objects with additional event 
+timestamp and some additional general information about request in `argumentsKw`. It can also be an empty array in case there were no publications to specified subscription yet or all 
+events were filtered out by specified criteria.
 
 {align="left"}
-        HELLO.Details.roles.<role>.features.event_history|bool := true
+```javascript
+  [
+    {
+        "timestamp": "yyyy-MM-ddThh:mm:ss:SSSZ", // Ыtring with event date in ISO-8601 format
+        "subscription": 2342423, // The subscription ID of the event
+        "publication": 32445235, // The original publication ID of the event
+        "details": {},           // The original details of the event
+        "args": [],              // The original list arguments payload of the event. May be ommited
+        "kwargs": {}             // The original key-value arguments payload of the event. May be ommited
+    }
+  ]
+```
+
+In cases when the events set is pretty big to send it in a single result, router implementations
+may provide additional options, like pagination or returning a progressive results. 
+
+As Event History feature operates on `subscription|id` there can be situations when there is no subscribers to topic
+of interest yet, but publications happens. In this case the *Broker* can not prematurely know what events to store.
+If the *Broker* implementation allows configuration on per-topic basis, it may overcome this situations by 
+preinitializing history-enabled topics with subscriptions even if there is no real subscribers yet exists.
+
+Sometimes a client may not be interested in subscribing to a topic just to get a subscription id. In that case
+a client may use some [Subscriptions Meta API RPC](#name-procedures-3) for retrieving subscription IDs by topic URIs
+if WAMP router supports it.
+
+**Security Aspects**
+
+To be able to request event history, peer must be allowed to subscribe to desired subscription first. Thus, if peer
```

Comment:
This is a crucial element which needs more discussion, so we could at least mark this here in the text to be merged in another follow up PR once we have it.

IMO, there are 2 valid approaches to authorization:

1. consider if the peer is authorized to subscribe _now_, and if so, allow the peer to access the complete history - so make it, as if the client was subscribed ever since
2. consider if the peer was authorized to subscribe in the _past_, or generally filter all results to those periods in which the peer was indeed allowed to subscribe

both are sufficient for a peer to catch up with missed events, however, the latter is more restrictive

Reply:
Yeah, agree. Added a todo/fixme note.